lib/services/downloads_service.dart

```diff
@@ -681,15 +683,56 @@ class DownloadsService {
     // Step 4 - Make sure there are no unanchored nodes in metadata.
     _downloadsLogger.info("Starting downloads repair step 4");
     downloadCounts[repairStepTrackingName] = 4;
+    final Map<int, LyricDto?> idsWithLyrics = HashMap();
+    var allItems = _isar.downloadItems.where().findAllSync();
```

Comment:
This can be pre-filtered to just songs which are not notDownloaded for a minor efficiency gain.

Reply:
like so?

Reply:
Yeah, that looks good.